src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/SymbolKey/SymbolKey.NamedTypeSymbolKey.cs

```diff
@@ -76,12 +80,32 @@ public static SymbolKeyResolution Resolve(SymbolKeyReader reader, out string? fa
                 bool isUnboundGenericType,
                 ITypeSymbol[] typeArguments)
             {
-                foreach (var type in container.GetTypeMembers(GetName(metadataName), arity))
+                var nameWithoutArity = removeArity(metadataName);
+                // Need to do a "decoding" step to get a file type source name from its metadata name
+                var sourceName = s_fileTypeNamePattern.Match(nameWithoutArity) is { Success: true, Length: var length }
+                    ? nameWithoutArity.Substring(length)
+                    : nameWithoutArity;
```

Comment:
is there no `.IsFile` symbol API yet?  i really don't like detecting based on random encoding details of these names.

Reply:
I'm not a fan either. We are in a position where we have a metadata name and need to get a source name in order to do a lookup. How can we do that sanely? I think that just knowing that `.IsFile` returned true isn't enough. We still have to figure out what source name to pass to `.GetTypeMembers` to get possible symbols to resolve the SymbolKey.

Reply:
> I'm not a fan either. We are in a position where we have a metadata name and need to get a source name in order to do a lookup. How can we do that sanely?

if we know it's a file-type, we can make a new key which encodes its filepath, containing symbol, name and arity.  Then,  resolution can resolve the containing symbool, and lookup the child type in that namespace that has that name/arity and which is located in the specified file.

Reply:
Have merged in the changes from #62218 and added the suggested public API.